src/storage/storage_backend.rs

```diff
@@ -160,6 +160,16 @@ pub trait StorageBackend<U: Sync + Send + Debug>: Send + Sync + Debug {
     /// [`Metadata`]: ./trait.Metadata.html
     async fn metadata<P: AsRef<Path> + Send + Debug>(&self, user: &Option<U>, path: P) -> Result<Self::Metadata>;
 
+    /// Returns the `Md5` for the given file.
+    ///
+    /// [`Metadata`]: ./trait.Md5.html
+    async fn md5<P: AsRef<Path> + Send + Debug>(&self, _user: &Option<U>, _path: P) -> Result<String>
+    where
+        P: AsRef<Path> + Send + Debug,
+    {
```

Comment:
Could work. We could also do a implementation where we call get and calculate the md5 on the stream

Reply:
Ah yeah, ofcourse. That would be a simple matter of moving the fs implementation here... Good suggestion! Why hadn't I thought of it.

Reply:
Done